pkg/operations/kubernetesupgrade/upgradecluster.go

```diff
@@ -101,10 +103,18 @@ func (uc *UpgradeCluster) UpgradeCluster(az armhelpers.AKSEngineClient, kubeConf
 		kubeClient = k
 	}
 
-	if err := uc.getClusterNodeStatus(kubeClient, uc.ResourceGroup); err != nil {
+	if err := uc.getClusterVMStatus(kubeClient, uc.ResourceGroup); err != nil {
 		return uc.Translator.Errorf("Error while querying ARM for resources: %+v", err)
 	}
 
+	if kubeClient != nil {
+		mastersCount := uc.DataModel.Properties.MasterProfile.Count
+		if err := uc.getClusterNodeStatus(kubeClient.ListNodesByOptions, mastersCount); err != nil {
```

Comment:
This call to the new `getClusterNodeStatus` method does not allow for any retries. Because a Node can be temporarily in a NotReady state, we want to be able to tolerate transient NotReady-ness. How are we going to do so?

Reply:
Right now it is not handled. Because all this happens before any changes to the resources/nodes state, the reasoning was to let a human characterize the failure and decide what to do about it. I can add a retry.

Reply:
I think it's fine, we don't need to add a retry just because. But FYI in case this actually reduces folks' ability to programmatically upgrade clusters (let's say you're using a script that quickly re-runs upgrade in a failure scenario)

Reply:
Added retry with a 2.5 mins timeout